### PR TITLE
tech: use `doctests-parallel`

### PR DESCRIPTION
**Pull request by ldicarlo**
Sorry for all the noise
Part of #46

### CODE REVIEW

**Comment by ldicarlo** [comment]
It looks better like that I think.

**Comment by ldicarlo** [comment]
```
haskus-binary       > [ERROR  ] [ThreadId 316] packages/haskus-binary/src/lib/Haskus/Number/Signed.hs:110: failure in expression `signedIsZero (signedNeg @5)'
haskus-binary       > [ERROR  ] [ThreadId 316] expected: False
haskus-binary       > [ERROR  ] [ThreadId 316]  but got:
haskus-binary       > [ERROR  ] [ThreadId 316]           ^
haskus-binary       > [ERROR  ] [ThreadId 316]           <interactive>:57:15: error:
haskus-binary       > [ERROR  ] [ThreadId 316]               Illegal visible type application ‘@5’
haskus-binary       > [ERROR  ] [ThreadId 316]                 Perhaps you intended to use TypeApplications
haskus-binary       > [ERROR  ] [ThreadId 316]
haskus-binary       > [ERROR  ] [ThreadId 316]           <interactive>:57:26: error:
haskus-binary       > [ERROR  ] [ThreadId 316]               Illegal type: ‘5’ Perhaps you intended to use DataKinds
```

I don't know how to fix these two issues

**Comment by ldicarlo** [comment]
@hsyl20 (or anyone), can you please guide me through the resolution of the error above? 
I tried to understand the code but it is above me.

**Comment by hsyl20** [comment]
You probably have to explicitly enable these instructions in the `$setup` code with:

```
> :set -XTypeApplications
> :set -XDataKinds
```

(not tested)

**Comment by ldicarlo** [comment]
> You probably have to explicitly enable these instructions in the `$setup` code with:
> 
> ```
> > :set -XTypeApplications
> > :set -XDataKinds
> ```
> 
> (not tested)

Of course ... I tried with `{-# ... #-}`  syntax instead...

File: haskus-binary/src/lib/Haskus/Binary/Serialize/Buffer.hs [hunk 1]
```diff
@@ -5,30 +5,28 @@
...
--- >>> xs == [0x01,0x23,0x45,0x67,0x89]
--- True
---
--- >>> b <- newBuffer 2 -- small buffer
--- >>> (_,b',_) <- runBufferPut b 0 overflowBufferDouble w
--- >>> xs <- forM [0..4] (bufferReadWord8IO b')
--- >>> xs == [0x01,0x23,0x45,0x67,0x89]
--- True
--- >>> bufferSize b'
--- 16
```
Comment (ldicarlo) [inline haskus-binary/src/lib/Haskus/Binary/Serialize/Buffer.hs:23]: I tried to remove those tests because I could not fix them

**Comment by thielema** [comment]
alternative could be doctest-extract

**Comment by hsyl20** [comment]
I've fixed the tests on CI with #55.

There is still a doctest failure in haskus-binary but I believe it's due to doctest or ghc.

**Pull request closed without merging**